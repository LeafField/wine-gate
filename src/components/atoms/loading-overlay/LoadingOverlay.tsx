import React, { FC } from "react";

const LoadingOverlay: FC = () => {
  return (
    <div className="overlay z-overlay animate-fadeIn">
      <div className="relative z-nav-menu h-8 w-8 animate-spin rounded-full border-4 border-gray border-t-transparent">
        <span className="sr-only">サーバーと通信中......</span>
      </div>
    </div>
  );
};

export default LoadingOverlay;
