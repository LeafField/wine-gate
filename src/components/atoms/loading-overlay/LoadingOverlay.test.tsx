import { screen, render } from "@testing-library/react";
import LoadingOverlay from "./LoadingOverlay";

describe("LoadingOverlayの単体テスト", () => {
  test("アクセシビリティ用のsr-onlyのテキストが存在する", () => {
    render(<LoadingOverlay />);
    const srOnly = screen.getByText("サーバーと通信中......");
    expect(srOnly).toBeInTheDocument();
  });

  test("スナップショットテスト", () => {
    const { container } = render(<LoadingOverlay />);
    expect(container).toMatchSnapshot();
  });
});
