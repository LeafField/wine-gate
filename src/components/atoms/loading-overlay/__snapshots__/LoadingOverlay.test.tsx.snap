// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlayの単体テスト スナップショットテスト 1`] = `
<div>
  <div
    class="overlay z-overlay animate-fadeIn"
  >
    <div
      class="relative z-nav-menu h-8 w-8 animate-spin rounded-full border-4 border-gray border-t-transparent"
    >
      <span
        class="sr-only"
      >
        サーバーと通信中......
      </span>
    </div>
  </div>
</div>
`;
